packages/opentelemetry-core/src/trace/sampler/AlwaysOnSampler.ts

```diff
@@ -14,16 +14,23 @@
  * limitations under the License.
  */
 
-import { Sampler, SamplingDecision, SamplingResult } from '@opentelemetry/api';
+import {
+  Context,
+  Sampler,
+  SamplingDecision,
+  SamplingResult,
+  trace,
+} from '@opentelemetry/api';
 
 /**
  * @deprecated Use the one defined in @opentelemetry/sdk-trace-base instead.
  * Sampler that samples all traces.
  */
 export class AlwaysOnSampler implements Sampler {
-  shouldSample(): SamplingResult {
+  shouldSample(context: Context): SamplingResult {
     return {
       decision: SamplingDecision.RECORD_AND_SAMPLED,
+      traceState: trace.getSpanContext(context)?.traceState,
```

Comment:
Returning a `SamplingResult` without `traceState` being set should indicate that the new span's trace state is set from the parent span's context. And it is actually handled in the `tracer.startSpan`.

So I think it is not necessary to set the trace state here.

Reply:
@legendecas @raphael-theriault-swi could this be addressed? This PR has been dragging on for a bit now and would be really nice to get it merged and released

Reply:
I should be able to address it some time today yes, been busy with other stuff until yesterday